neo.UnitTests/SmartContract/UT_Syscalls.cs

```diff
@@ -1,66 +1,120 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
 using Neo.SmartContract;
 using Neo.VM;
+using Neo.VM.Types;
 using System.Linq;
 
 namespace Neo.UnitTests.SmartContract
 {
     [TestClass]
     public class UT_Syscalls
     {
+        [TestMethod]
+        public void System_ExecutionEngine_GetScriptContainer()
+        {
+            var snapshot = TestBlockchain.GetStore().GetSnapshot();
+            using (var script = new ScriptBuilder())
+            {
+                script.EmitSysCall(InteropService.System_ExecutionEngine_GetScriptContainer);
+
+                // Without tx
+
+                var engine = new ApplicationEngine(TriggerType.Application, null, snapshot, 0, true);
+                engine.LoadScript(script.ToArray());
+
+                Assert.AreEqual(engine.Execute(), VMState.HALT);
+                Assert.AreEqual(1, engine.ResultStack.Count);
+                Assert.IsNull(((InteropInterface<IVerifiable>)engine.ResultStack.Pop()).GetInterface<IVerifiable>());
+
+                // With tx
+
+                script.EmitSysCall(InteropService.Neo_Json_Serialize);
+
+                var tx = new Transaction()
+                {
+                    Script = new byte[] { 0x01 },
+                    Attributes = new TransactionAttribute[0],
+                    Cosigners = new Cosigner[0],
+                    NetworkFee = 0x02,
+                    SystemFee = 0x03,
+                    Nonce = 0x04,
+                    ValidUntilBlock = 0x05,
+                    Version = 0x06,
+                    Witnesses = new Witness[] { new Witness() { VerificationScript = new byte[] { 0x07 } } },
+                    Sender = UInt160.Parse("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"),
+                };
+
+                engine = new ApplicationEngine(TriggerType.Application, tx, snapshot, 0, true);
+                engine.LoadScript(script.ToArray());
+                engine.CurrentContext.EvaluationStack.Push(InteropInterface.FromInterface(tx));
+
+                Assert.AreEqual(engine.Execute(), VMState.HALT);
+                Assert.AreEqual(2, engine.ResultStack.Count);
+                Assert.AreEqual(engine.ResultStack.Pop().GetString(),
+                    @"[""\uFFFDD\uFFFDa\uFFFDs,\uFFFD\uFFFDf\u0007]\u0000\uFFFD\u0622\uFFFDT\uFFFD7\u00133\u0018\u0003e\uFFFD\uFFFD\u0027Z\uFFFD\/\uFFFD"",""\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\uFFFD"",""\u0001"",[""\u0007""],4,2,3,5,6]");
+                Assert.AreEqual(1, engine.ResultStack.Count);
```

Comment:
@erikzhang `jsonSerialize` should pop the serialized item, isn't it? currently is leaving in the stack

Reply:
Pop the unserialized item, and push the serialized item, no?

Reply:
Should be like this, i think, but both are in the stack

Reply:
Was my fault in the unit test, sorry, i push the transaction twice